crates/uv/tests/it/sync.rs

```diff
@@ -1273,6 +1273,143 @@ fn sync_non_existent_group() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn sync_non_existent_default_group() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = ["typing-extensions"]
+
+        [dependency-groups]
+        foo = []
+
+        [tool.uv]
+        default-groups = ["bar"]
+        "#,
+    )?;
+
+    context.lock().assert().success();
+
+    uv_snapshot!(context.filters(), context.sync(), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: Default group `bar` is not defined in the project's `dependency-group` table
```

Comment:
Perhaps something like

```suggestion
    error: Default group `bar` (from `tool.uv.default-groups`) is not defined in the project's `dependency-group` table
```

Reply:
Done